src/Options/Applicative/Complicated.hs

```diff
@@ -37,14 +41,14 @@ complicatedOptions
   -- ^ program description (displayed between usage and options listing in the help output)
   -> String
   -- ^ footer
-  -> Parser a
+  -> Parser GlobalOptsMonoid
   -- ^ common settings
-  -> Maybe (ParserFailure ParserHelp -> [String] -> IO (a,(b,a)))
+  -> Maybe (ParserFailure ParserHelp -> [String] -> IO (GlobalOptsMonoid,(RIO env (),GlobalOptsMonoid)))
   -- ^ optional handler for parser failure; 'handleParseResult' is called by
   -- default
-  -> ExceptT b (Writer (Mod CommandFields (b,a))) ()
+  -> AddCommand env
```

Comment:
If moving from polymorphic to concrete, I wondered if `env` was always `Runner`, in practice.